File: single-language-analysis/dataset-files/python/test_smell_011.py
import pytest

class TestTaskTest:
    def test_smell_011(self):
        error = ""
        error += ("test_task_to_string() \t Task toString\n")
        
        t1 = Task("T1", 1)
        ans = "\tTask: T1\tLevel: 1"
        students = t1.to_string()
        
        min_length = min(len(ans), len(students))
        if len(ans) != len(students):
            error += ("\tSize was " + str(len(ans)) + " but received size " + str(len(students)) + "\n\n")
        
        for i in range(min_length):
            if ans[i] != students[i]:
                temp1 = ""
                temp2 = ""

                if ans[i] == '\t':
                    temp1 = "\\t"
                elif ans[i] == '\n':
                    temp1 = "\\n"
                else:
                    temp1 = ans[i]

                if students[i] == '\t':
                    temp2 = "\\t"
                elif students[i] == '\n':
                    temp2 = "\\n"
                else:
                    temp2 = students[i]
                error += ("\tIndex: " + str(i) + " Ans: " + temp1 + " Students: " + temp2 + "\n")    
        
        error += ("\n")
        
        assert ans == students

File: single-language-analysis/dataset-files/java/TestSmell011.java
// Original URL: https://github.com/chadwishner/COP3502-Project-4-Java-/blob/d2a79f7fc4a9883f68f60744cb2866225fe449ec/TaskTest.java#L10-L54

public class TestSmell011 {
	@Test
	public void testTaskToString(){
		error += ("testTaskToString() \t Task toString\n");
		
		Task t1 = new Task("T1", 1);
		String ans = "\tTask: T1\tLevel: 1";
		String students = t1.toString();
		
		int min = Math.min(ans.length(), students.length());
		if(ans.length() !=  students.length()){
			error += ("\tSize was " + ans.length() + " but received size " + students.length() + "\n\n");
		}
		
		
		for(int i = 0; i < min; i++){
			if(ans.charAt(i) != students.charAt(i)){
				String temp1 = "";
				String temp2 = "";

				if(ans.charAt(i) == '\t'){
					temp1 = "\\t";
				}
				else if(ans.charAt(i) == '\n'){
					temp1 = "\\n";
				}
				else{
					temp1 = ("" + ans.charAt(i));
				}

				if(students.charAt(i) == '\t'){
					temp2 = "\\t";
				}
				else if(students.charAt(i) == '\n'){
					temp2 = "\\n";
				}
				else{
					temp2 = ("" + students.charAt(i));
				}
				error += ("\tIndex: " + i + " Ans: " + temp1 + " Students: " + temp2 + "\n");	
			}
		}
		error += ("\n");
		
		assert(ans.equals(students));
	}
}
